### DISCUSSION TITLE
科技新聞分享—為什麼軟體工程師面試時，最好不要現場 Coding？

### DISCUSSION THREAD

**Discussion by nsda890750** [1101-GS3073-C/Python-Community-GS3073-C#319] [科技新知]
第一，測驗的題目可能會在網路上流傳，而認真準備的面試者可能可以透過練習記住答案；第二，這樣的測驗方式可能給面試者帶來很大的壓力，導致表現失常；第三，這樣的測驗其實並不符和日常工作方式，因為大多數的工作者在 coding 的過程中，可以透過 google 找到更好的方法。

較推薦的軟體工程師面試方法：

國外網站建議，與其讓面試者寫程式，不如考慮讓他們閱讀現有程式並說明作用和原理。這樣的方式有以下幾項優點：

1. 可以看出面試者的最基本技能

閱讀程式可能佔了開發人員工作的 95%，無論開發人員是在寫新的程式或是除 bug ，他們都需要不斷地重複閱讀。

2. 讀程式比寫程式更有效率

面試官可以在面試者開始說明 code 的五分鐘內，就大概明白面試者的程度。因為閱讀與口語表達所需要的時間比寫程式少很多，讓面試官可以有更多時間去深入了解面試者。

3. 閱讀讓面試者比較輕鬆

面試官其實不應該給予面試者過高的壓力，因為壓力會提高腎上腺素，進而降低面試者思考的能力，導致最後可能錯過優秀的候選人。許多面試者也提到，在面試中進行 coding 測驗時，考官們專注的檢視每一次輸入是很大的壓力。

這些面試者因不必 coding 而比較不會有壓力，面試官也可以隨時更換適合候選人的問題以及技能。

資料連結：
https://buzzorange.com/techorange/2022/04/28/find-great-developers-by-having-them-read-code/

**Comment by number32984** [comment]
學廢了，感恩!

**Comment by 32844583** [comment]
獲益良多，不過有些公司還是會以其他方式要求現場 coding 吧，況且只讀程式很難記住整個函式的原理，還是推薦平時多多實作好

**Comment by brian431** [comment]
Homebrew 的作者就曾因為現場coding沒有表現好而被Google刷掉。這件事情也引起了許多人討論這種白板面試是否真的能代表應徵者的開發能力。

**Comment by NcuJerryKao** [comment]
這讓我想到 高中老師不太會打程式 但很會查資料 找到相應的問題解決方法
就像整合開發環境一樣 優雅

**Comment by BingRong1217** [comment]
原來現場測試coding有那麼多缺點

**Comment by xCk27x** [comment]
雖然這樣的面試方法會有對實作能力的質疑
但其實從對一段程式碼的了解中
就能判斷面試者大部分的實作能力了~
因為很多細節一定要透過實作出錯過才會發現到

**Comment by Ben6uo** [comment]
但這樣就不知道面試人的程式能力了嗎?

**Comment by limitx0** [comment]
我也覺得現場coding不是很好的面試方法
和平常在開發的過程差太多，需要的能力也很不一樣

**Comment by awehao** [comment]
我覺得打出程式，跟如何解讀和乳和簡化比，後者需要的不只是能打出程式的實力，需要更多技能

**Comment by skyling0299** [comment]
可是事實上白版題想考得不只是腦筋急轉彎，而是想看看受試者如何思考。透過即時的分享和回饋讓大腦邊做邊想，解釋自己寫法的脈絡，甚至是現場debug，都是能讓團隊看出一個人coding的習慣的方法

**Reply by RogelioKG** [reply]
所以我認為白板題不應該考太有難度的題目。
大型專案通常都是一個團隊在開發，一個合格的程式設計師應該要有能將自己想法有邏輯地表達出來的能力。在面試的過程中，應該要著重在受試者的表達過程和邏輯思維，即使是簡單的題目，也能從受試者的回答中推測出他對這項技術的了解程度與深度。

**Comment by Xuewen628** [comment]
我同意ㄟ coding這件事通常不會是由一個人完成再到完善的 其中需經由很多人的腦力激盪、靈光一閃 和我們解數學題不一樣 coding不只有一個解

**Comment by larry40305** [comment]
我覺得現場coding真的會非常的有壓力，很容易失常

**Comment by jcpan821** [comment]
我也比較傾向讀程式，但debug真的很累

**Comment by boson13579** [comment]
現場上機考真的有很多變數，以些看過或寫過類似考題的考生相對其他人更有優勢，比較像是在比運氣。
另外，上機考不開放Google搜尋也讓我質疑，因為真正工作時，工程師往往需要通過不斷查詢才能更精進程式碼，考試時卻要求不能搜尋???

**Comment by Yilin20021016** [comment]
我覺得有可能當天狀態不好 而打不出來 而且平時都會上google查

**Comment by liangyinglly** [comment]
比起現場上陣，很多可以上網查的作業或報告對我來說，我會學到的東西比較多，也比較能整理成比較完美的樣子呈現給打家看

**Comment by Vincent20030314** [comment]
感謝分享，讓我對未來面試的準備方向更加明確。

**Comment by HsuChen88** [comment]
之後自動產生code和debug的輔助工具都發展成熟之後，寫出怎樣的程式就不是人的事了，所以我覺得面試應該是要看出這個人如何設計架構，而不是著重在debug上。

**Comment by yjpxi** [comment]
傳統的寫程式測驗可能會讓面試者感到不必要的壓力，並且可能不完全反映他們在真實工作中的表現。相反，讓面試者閱讀現有程式並解釋其運作原理，更能測試他們的理解能力和基本技能，這在實際工作中是更為重要的。
